src/EntityFramework6.Npgsql/NpgsqlTextFunctions.cs

```diff
@@ -368,5 +369,38 @@ public static string TsRewrite(string query, string target, string substitute)
         {
             throw new NotSupportedException();
         }
-    }   
+
+        /// <summary>
+        /// Matches regular expression, case insensitive. Generates the "~*" operator.
+        /// http://www.postgresql.org/docs/current/static/functions-matching.html#FUNCTIONS-POSIX-REGEXP
+        /// </summary>
+        /// <param name="input">A string to search for a match.</param>
+        /// <param name="pattern">A regular expression pattern to match.</param>
+        /// <returns>
+        /// <see langword="true"/> if <paramref name="input"/> matches <paramref name="pattern"/>,
+        /// otherwise, <see langword="false"/>.
+        /// </returns>
+        [DbFunction("Npgsql", "regex_is_match")]
+        public static bool RegexIsMatch(string input, string pattern)
+        {
+            throw new NotSupportedException();
+        }
+
+        /// <summary>
+        /// Matches regular expression, case sensitive. Generates the "~" operator.
+        /// http://www.postgresql.org/docs/current/static/functions-matching.html#FUNCTIONS-POSIX-REGEXP
+        /// </summary>
+        /// <param name="input">A string to search for a match.</param>
+        /// <param name="pattern">A regular expression pattern to match.</param>
+        /// <param name="options">A bitwise combination of the enumeration values that provide options for matching.</param>
```

Comment:
This should state that ```RegexOptions.RightToLeft``` and ```RegexOptions.ECMAScript``` are unsupported to prevent consumers of this code from getting surprised. (As-is, people would find out at runtime or by reading the source).